File: trash/makecode.py
def on_data_received():
    pass
serial.on_data_received(serial.delimiters(Delimiters.HASH), on_data_received)

led.enable(False)

def on_forever():
    pins.digital_write_pin(DigitalPin.P0, 1)
    NPNBitKit.dht11_read(DigitalPin.P3)
    radio.send_string("!1:TEMP:" + str(NPNBitKit.dht11_temp()) + "#")
    pins.digital_write_pin(DigitalPin.P0, 0)
    basic.pause(3000)
basic.forever(on_forever)

File: trash/makecode.js
let cmd = ""
serial.onDataReceived(serial.delimiters(Delimiters.Hash), function () {
    cmd = serial.readUntil(serial.delimiters(Delimiters.Hash))
    if (cmd == "LTrue") {
        basic.showString("L1")
    } else if (cmd == "LFalse") {
        basic.showString("L0")
    } else if (cmd == "PTrue") {
        basic.showString("P1")
    } else if (cmd == "PFalse") {
        basic.showString("P0")
    }
})
basic.forever(function () {
    serial.writeString("!1:TEMP:" + input.temperature() + "#")
    basic.pause(5000)
    serial.writeString("!1:LIGHT:" + input.lightLevel() + "#")
    basic.pause(5000)
})
